charms/kfp-metadata-writer/tests/unit/test_operator.py

```diff
@@ -54,26 +54,52 @@ def test_grpc_relation_with_data(harness, mocked_lightkube_client):
     harness.charm.leadership_gate.get_status = MagicMock(return_value=ActiveStatus())
 
     # Add relation with data.  This should trigger a charm reconciliation due to relation-changed.
-    add_sdi_relation_to_harness(harness, "grpc", data=MOCK_GRPC_DATA)
+    rel_id = harness.add_relation(
+        relation_name=GRPC_RELATION_NAME, remote_app="other-app", app_data=MOCK_GRPC_DATA
+    )
 
     # Assert
-    assert isinstance(harness.charm.grpc_relation.status, ActiveStatus)
+    assert harness.charm.grpc_relation.component.get_service_info().name == MOCK_GRPC_DATA["name"]
+    assert harness.charm.grpc_relation.component.get_service_info().port == MOCK_GRPC_DATA["port"]
+
+    # Duplicity to check the same data in the relation is in the grpc_relation component
+    rel_data = harness.get_relation_data(relation_id=rel_id, app_or_unit="other-app")
+    assert harness.charm.grpc_relation.component.get_service_info().name == rel_data["name"]
+    assert harness.charm.grpc_relation.component.get_service_info().port == rel_data["port"]
```

Comment:
I think we should remove this check.  This tests Juju/harness behaviour, not something we created.  the above test of `get_service_info() == MOCK_GRPC_DATA` completely tests your code

Reply:
Updated.